_posts/2020-09-29-lldb-reproducers.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+title: "LLDB Reproducers"
+date: 2020-09-29 19:39
+---
+
+Swift developers love to complain about LLDB. While there are many
+reasaonable complaints, the important question is what can we do to make
+it better.
+
+Enter [reproducers][reproducers]. Reproducers provide a way to run LLDB
+while also capturing information about your debugging session. Then you
+can [submit][radar] the session to Apple to reproduce the bug you've
+experienced.
+
+# How?
+
+While the steps to use reproducers are mostly straight forward, the
+problem is launching LLDB from Xcode does not enable `--capture` mode
+(FB7878562). This means if you want to provide a reproducer for an issue
+you've experienced in a Xcode debugging session you need to reproduce
+it outside of Xcode instead.
+
+Note: to provide a useful reproducer, LLDB bundles all files the
+debugging session touched. This will include binaries with debug info
+that you may consider sensitive. Be sure to verify what you're sharing
+with Apple before you [send it][radar].
+
+
+## CLI / macOS app
+
+If you're debugging a program on your Mac, there are a few steps:
+
+1. Run the app in Xcode and stop it (so you know it's up to date).
+2. In Terminal.app navigate to your DerivedData directory (you can find
+   this by right clicking on your app in the "Products" section of
+   Xcode's project navigator, and clicking "Show in Finder").
+3. Run `lldb --capture /path/to/Your.app`.
+4. In the LLDB session run `process launch --stop-at-entry`.
+5. Now you're in a paused LLDB session. Here you can set whatever
+   breakpoints you need to reproduce your issue. Often for me this means
```

Comment:
Maybe worth linking to some docs on the various commands to set these? It's mentioned below but someone might not even get that far if they get stuck here.